### PR TITLE
fix: keep notification and heartbeat records when import backup with …

### PR DESCRIPTION
**Pull request by SunnyDoraemon**
…overwrite mode

⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes [#677](https://github.com/louislam/uptime-kuma/issues/677)

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by SunnyDoraemon** [comment]
Is this PR has any problem there?,should I cancel it?

**Comment by chakflying** [comment]
What you implemented is not only a fix (which there was already a PR open), but also a breaking change to the behavior of overwrite mode without previous discussion.

**Comment by Saibamen** [comment]
Related PR (already merged to fix https://github.com/louislam/uptime-kuma/issues/677): https://github.com/louislam/uptime-kuma/pull/2871

File: server/server.js [hunk 1]
```diff
@@ -1247,19 +1247,38 @@ let needSetup = false;
...
                     await R.exec("DELETE FROM monitor_tag");
                     await R.exec("DELETE FROM tag");
-                    await R.exec("DELETE FROM monitor");
+                    // for keep heartbeat record.
+                    // now keep monitors which in import backup list.(just for keep heartbeat records, because heartbeat table ddl set `ON DELETE CASCADE` )
+                    let monitorsIdList = await R.getAll("SELECT id FROM monitor");
+                    monitorsIdList = monitorsIdList.map((item) => {
+                        return item.id;
+                    });
+                    log.info("manage", "Import Backup with 'overwrite mode. now remove the monitors which is not in backup json'");
```
Comment (Saibamen) [inline server/server.js:1268]:
```suggestion
                    log.info("manage", "Import Backup with 'overwrite mode'. Now remove the monitors which is not in backup JSON'");
```

File: server/server.js [hunk 2]
```diff
@@ -1247,19 +1247,38 @@ let needSetup = false;
...
+                    // filter monitors which is not in backup monitorList
+                    for (let i = 0; i < monitorListData.length; i++) {
+                        let backupMonitorId = monitorListData[i].id;
+                        let idx = monitorsIdList.indexOf(backupMonitorId);
+                        if (idx > -1) {
+                            needOverWriteMonitorsIdList.push(backupMonitorId);
+                            monitorsIdList.splice(idx, 1);
+                        }
+                    }
+                    log.info("manage", `Need OverWrite Monitors Id List:[${needOverWriteMonitorsIdList}],Length:${needOverWriteMonitorsIdList.length}`);
```
Comment (Saibamen) [inline server/server.js:1278]:
```suggestion
                    log.info("manage", `Need OverWrite Monitors Id List: [${needOverWriteMonitorsIdList}], Length:${needOverWriteMonitorsIdList.length}`);
```

File: server/server.js [hunk 3]
```diff
@@ -1370,6 +1401,26 @@ let needSetup = false;
...
+                                log.info("manage", `[BackUp OverWrite Mode] Find monitor which need keep heartbeats records - Monitor ID: ${monitorListData[i].id}`);
+                                // relink monitors heartbeat records
+                                await R.exec("UPDATE `heartbeat` SET monitor_id = ? WHERE monitor_id = ? ", [
+                                    bean.id,
+                                    monitorListData[i].id
+                                ]);
+                                // delete old same monitor item with expired id
+                                await R.exec("DELETE FROM monitor where id= ? ", monitorListData[i].id);
+                            } else {
+                                log.info("manage", `[BackUp OverWrite Mode] Find new monitor Join - name:${monitorListData[i].name}`);
```
Comment (Saibamen) [inline server/server.js:1417]:
```suggestion
                                log.info("manage", `[BackUp OverWrite Mode] Find new monitor Join - name: ${monitorListData[i].name}`);
```

**Comment by chakflying** [comment]
Obsolete as the backup feature has been dropped in #3892.

**Pull request closed without merging**